test/k8sT/Chaos.go

```diff
@@ -0,0 +1,156 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("K8sChaosTest", func() {
+
+	var kubectl *helpers.Kubectl
+	var logger *logrus.Entry
+	var once sync.Once
+	var demoDSPath string
+	var ciliumPath string
+
+	initialize := func() {
+		logger = log.WithFields(logrus.Fields{"testName": "K8sChaosTest"})
+		logger.Info("Starting")
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+		ciliumPath = kubectl.ManifestGet("cilium_ds.yaml")
+		kubectl.Apply(ciliumPath)
+
+		_, err := kubectl.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", 600)
+		Expect(err).Should(BeNil())
+
+		err = kubectl.WaitKubeDNS()
+		Expect(err).Should(BeNil())
+
+		demoDSPath = kubectl.ManifestGet("demo_ds.yaml")
+	}
+
+	BeforeEach(func() {
+		once.Do(initialize)
+		kubectl.Apply(demoDSPath).ExpectSuccess("DS deployment cannot be applied")
+
+		_, err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l zgroup=testDS"), 300)
+		Expect(err).Should(BeNil(), "Pods are not ready after timeout")
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			ciliumPod, _ := kubectl.GetCiliumPodOnNode("kube-system", "k8s1")
+			kubectl.CiliumReport("kube-system", ciliumPod, []string{
+				"cilium service list",
+				"cilium endpoint list"})
+		}
+
+		kubectl.Delete(demoDSPath).ExpectSuccess("DS deployment cannot be applied")
+		err := kubectl.WaitCleanAllTerminatingPods()
+		Expect(err).To(BeNil(), "Terminating containers are not deleted after timeout")
+
+	})
+
+	PingToService := func(ctx context.Context) {
+		// Pings from pods clients to services in background meanwhile context
+		// is not terminated.
+		go func() {
+			defer GinkgoRecover()
+			pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, "zgroup=testDSClient")
+			Expect(err).To(BeNil(), "Cannot get pods names")
+			Expect(len(pods)).To(BeNumerically(">", 0), "No pods available to test connectivity")
+
+			host, port, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, "testds-service")
+			Expect(err).To(BeNil(), "cannot get the service")
+			service := fmt.Sprintf("http://%s", net.JoinHostPort(host, fmt.Sprintf("%d", port)))
+
+			for {
+				select {
+				case <-ctx.Done():
+					logger.Info("Cancel event in PingToService")
+					return
+				default:
+					for _, pod := range pods {
+						_, err := kubectl.ExecPodCmd(
+							helpers.DefaultNamespace, pod, helpers.CurlFail(service))
+						if err != nil {
```

Comment:
Why log to the logger if the error is nil but then have the Expect to check for the nil error immediately after? Seems redundant to me.